src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.ComposeCache.targets

```diff
@@ -23,11 +23,127 @@ Copyright (c) .NET Foundation. All rights reserved.
   </PropertyGroup>
 
   <Target Name="ComposeCache"
-          DependsOnTargets="PrepareForComposeCache;
-                            RestoreForComposeCache;
-                            ComputeAndCopyFilesToCacheDirectory"/>
+          DependsOnTargets="_InitializeBasicProps;
+                            PrepareForComposeCache;
+                            PrepOptimizer;
+                            CacheWorkerMain;
+                            CacheFinalizer;"/>
+
+
+  <!--
+    ============================================================
+                                        CacheWorkerMain
+
+   Spins of the compose mappers
+    ============================================================
+    -->
+
+  <Target Name="CacheWorkerMain">
+    
+    <ItemGroup>
+      <_AllProjects Include="$(AdditionalProjects.Split('%3B'))"/>
+      <_AllProjects Include ="$(MSBuildProjectFullPath)"/>
+    </ItemGroup>
+    
+    <MSBuild Projects="%(_AllProjects.Identity)"
```

Comment:
We can do a lot of this work and the CacheWorkerMapper work in parallel, right?  Or does it have to be sequential?